Source/loadsave.cpp

```diff
@@ -93,9 +95,12 @@ class LoadHelper {
 	}
 
 public:
-	LoadHelper(const char *szFileName)
+	LoadHelper(std::optional<MpqArchive> archive, const char *szFileName)

```

Comment:
I'm not super happy about it, but I feel resolving it would require a lot of general refactoring to how save games are handled. So going to leave it like this for now.